Client/Frontend/Browser/UserScriptManager.swift

```diff
@@ -363,6 +363,31 @@ class UserScriptManager {
       forMainFrameOnly: true,
       in: .defaultClient)
   }()
+  
+  private let ReadyStateScript: WKUserScript? = {
+    guard let path = Bundle.main.path(forResource: "ReadyState", ofType: "js"), let source = try? String(contentsOfFile: path) else {
+      log.error("Failed to load ReadyState.js")
+      return nil
+    }
+
+    var alteredSource = source
+    let token = UserScriptManager.securityTokenString
+
+    let replacements = [
+      "$<security_token>": token,
+      "$<handler>": "ReadyState_\(messageHandlerTokenString)",
+    ]
+
+    replacements.forEach({
+      alteredSource = alteredSource.replacingOccurrences(of: $0.key, with: $0.value, options: .literal)
```

Comment:
I don't know if in Swift this ends up alloc'ing multiple copies of the same inserted script, but you can get the same behavior without needing the find / replace or string operations by doing the approach @cuba used here:
https://github.com/brave/brave-ios/pull/5156/files#diff-a059035e1311558281829f11147315c797da830d78176db17a5515210e98e1a3R72

Anyway, not a blocking issue, but perf and code cleanliness suggestion (since you always have valid JS)

Reply:
The file linked above does do a `find and replace` operation.
If you mean caching, I will be doing it this way in a future PR: https://github.com/mozilla-mobile/firefox-ios/blob/main/Client/Frontend/TabContentScripts/UserScriptManager.swift#L22

Reply:
I think we might be talking about different things; im referring to the technique @cuba does to avoid having to splice up JS code for the farbling.  Client/Frontend/UserContent/UserScripts/FarblingProtection.js is a complete JS implementation, and then he just concatenates an invocation with it at the location i linked to.

I'm suggesting this technique for two reasons:
- It ensures proper JS is in the repo, so it can be linted / checked / etc
- it (might, depending on what Swift does under the hood) prevents the need to keep multiple copies of the JS code in memory, since you don't need a copy per copy/replace operation

Reply:
The reason `FarblingProtection` is a complete JS implementation is because it doesn't have to replace the `SecurityToken` in the JS. That's because it doesn't have any callbacks to Swift Code. IE: It doesn't pass any information from JS to Swift. My code passes information from JS to Swift, and so in order to prevent pages from accessing the message handlers directly, we have a security token replacement done at runtime, as this cannot be sandboxed.

Same code as here: https://github.com/brave/brave-ios/pull/5156/files#diff-a059035e1311558281829f11147315c797da830d78176db17a5515210e98e1a3R87

Reply:
I understand, but why implement your injected script as a function, and then pass the security token as a parameter to the function (this is what the linked to code does, it handles the farbling values in the same way your code could handle the security token

Reply:
Like this https://gist.github.com/pes10k/fb6041957406b11f81188c18e6361c2a

And then inject `alteredSource` + "window.bravePageEvents(\(<swift_security_token>), \(<swift_handler_name));delete window.bravePageEvents";

Reply:
Why would I do that though? That requires me to `evaluateJavascript` every time the script is injected. That means my script won't be able to run `onDocumentStart` which is done automatically by WebKit.

If I do it your way, I have to do `webKit.evaluateSafeJavascript(function: "window.braveEvents", args: ["securityToken", ...])` somewhere in the code. Possibly when the page is already finished loading. Plus this is also `async`, so I may not get the event at all.

Reply:
Once we switch over to the code I linked from Firefox for ALL the scripts, it'd be done better than that:
`let wrappedSource = "(function() { const SECURITY_TOKEN = '\(UserScriptManager.securityToken)'; \(source) })()"`

So we won't have to execute a function and then delete it right after.

Reply:
To my mind, that still seems likely to wind up with a bunch of large strings one-off strings that aren't cacheable.  I dont understand why thats preferable, but its also not a privacy concern so im fine dropping the issue and closing the privacy review

Reply:
At least in some languages that would not be true, in terms of number of allocs and memory used.  I don't know how swift would handle that kind of string concatenation or interpolation, so maybe im generalizing incorrectly.  I'm more than happy to defer to your Swift knowledge here and drop this concern